multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/jobs/LockOwnerCleaner.java

```diff
@@ -0,0 +1,51 @@
+package org.cloudfoundry.multiapps.controller.process.jobs;
+
+import org.cloudfoundry.multiapps.controller.persistence.model.LockOwnerEntry;
+import org.cloudfoundry.multiapps.controller.persistence.services.LockOwnerService;
+import org.flowable.engine.ProcessEngine;
+import org.flowable.engine.impl.cmd.ClearProcessInstanceLockTimesCmd;
+import org.springframework.scheduling.annotation.Scheduled;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+@Named
+public class LockOwnerCleaner {
+
+    private final ProcessEngine processEngine;
+    private final LockOwnerService lockOwnerService;
+
+    @Inject
+    public LockOwnerCleaner(ProcessEngine processEngine, LockOwnerService lockOwnerService) {
+        this.processEngine = processEngine;
+        this.lockOwnerService = lockOwnerService;
+    }
+
+    @Scheduled(fixedRate = 6, timeUnit = TimeUnit.MINUTES)
+    public void run() {
+        var timestamp = LocalDateTime.now();
+        var staleLockOwners = lockOwnerService.createQuery()
+                                              .olderThan(timestamp.minus(Duration.ofMinutes(6)))
+                                              .list()
+                                              .stream()
+                                              .map(LockOwnerEntry::getLockOwner)
+                                              .collect(Collectors.toList());
+        for (var staleLockOwner : staleLockOwners) {
+            //is it possible to have a race condition here?
+            //this job should only do a transaction to update Flowable's tables, so
+            //there shouldn't be a race condition, in theory
+            processEngine.getProcessEngineConfiguration()
+                         .getCommandExecutor()
+                         .execute(new ClearProcessInstanceLockTimesCmd(staleLockOwner)); //what if this throws error?
```

Comment:
Are these comments helpful?

Also if it can throw an exception, catch it and log it

Reply:
Log what was cleared: staleLockOwners